docs/designers-developers/developers/data/data-core-block-editor.md

```diff
@@ -233,7 +233,7 @@ _Parameters_
 
 _Returns_
 
--   `Array<WPBlock>`: Block objects.
+-   `null`: Block objects.
```

Comment:
@nosolosw - it looks like `import` statements for type defintions gets converted into `null` by `npm run docs:build`

Reply:
It looks like a known issue: https://github.com/WordPress/gutenberg/issues/15186. Can we at least remove the import part and keep the rest as is?

Reply:
That import syntax seems to originate from TypeScript and is supported by a few IDEs as well as `eslint-plugin-jsdoc`

Reply:
It looks like `eslint-plugin-jsdoc` uses `jsdoctypeparser`, which includes support for other syntaxes (typescript, etc). We are due for migrating `docgen` out of `doctrine` (the JSDoc parser), so `jsdoctypeparser` could be a good replacement. Created an issue with some info at https://github.com/WordPress/gutenberg/issues/18045

Note that I can't commit to this migration in the next month (perhaps by the end of November I could), but happy to assist anyone that is up the game.

Reply:
Do you have an idea about whether we could provide a temporary fix to move forward with ESLint warnings removal?

Reply:
Not sure what can be a temporary fix: getting structured information out of the JSDoc comment is what `doctrine` does and it returns `null` for extraneous types. From that first step, `docgen` links the given type to the exported artifact, so it'll carry `null` all the way up. The only temporary fix I can think of is to avoid using this kind of syntax until our tooling has support.

Reply:
How much work do you anticipate for `jsdoctypeparser` integration in place of `doctrine`?